### PR TITLE
ansilove 3.0.8 (new formula)

### PR DESCRIPTION
**Pull request by ghost**
- [ ✅ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ ✅ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ✅ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ✅ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
New formula for the ansilove project.  I noticed that this package is easily available on freebsd / linux so I thought I would add it to homebrew.

### CODE REVIEW

File: Formula/ansilove.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  homepage "https://www.ansilove.org"
+  url "https://github.com/ansilove/ansilove/archive/3.0.8.tar.gz"
+  sha256 "655472d31a9ec5a5ba29f5c15485d8647526ff8b0c9968d71f4267a9aa9bb544"
+  depends_on "cmake" => :build
+  depends_on "gd"
+
+  def install
+    system "cmake", "."
+    system "make", "ansilove"
+    libexec.install "ansilove"
```
Comment (fxcoudert) [inline Formula/ansilove.rb]: Is there no `make install` target?

File: Formula/ansilove.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    bin.install_symlink libexec/"ansilove"
+  end
+
+  test do
+    system "#{bin}/ansilove", "-e"
```
Comment (fxcoudert) [inline Formula/ansilove.rb]: What does this test do?

File: Formula/ansilove.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  url "https://github.com/ansilove/ansilove/archive/3.0.8.tar.gz"
+  sha256 "655472d31a9ec5a5ba29f5c15485d8647526ff8b0c9968d71f4267a9aa9bb544"
+  depends_on "cmake" => :build
+  depends_on "gd"
+
+  def install
+    system "cmake", "."
+    system "make", "ansilove"
+    libexec.install "ansilove"
+    bin.install_symlink libexec/"ansilove"
```
Comment (fxcoudert) [inline Formula/ansilove.rb]: This should be directly install in `bin`

File: Formula/ansilove.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+class Ansilove < Formula
+  desc "ANSi / ASCII art to PNG converter in C"
+  homepage "https://www.ansilove.org"
+  url "https://github.com/ansilove/ansilove/archive/3.0.8.tar.gz"
+  sha256 "655472d31a9ec5a5ba29f5c15485d8647526ff8b0c9968d71f4267a9aa9bb544"
+  depends_on "cmake" => :build
```
Comment (fxcoudert) [inline Formula/ansilove.rb]: Blank link before first `depends_on`

File: Formula/ansilove.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+class Ansilove < Formula
+  desc "ANSi / ASCII art to PNG converter in C"
+  homepage "https://www.ansilove.org"
```
Comment (fxcoudert) [inline Formula/ansilove.rb]: Trailing slash, please

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/ansilove.rb (hunk 1 above)
Comment (usbsnowcrash) [reply Formula/ansilove.rb]: Looks like the default make target will work so I am switching to that

File: Formula/ansilove.rb (hunk 2 above)
Comment (usbsnowcrash) [reply Formula/ansilove.rb]: Switched to to -v which outputs version info.  I am unsure what else would be an acceptable test?

File: Formula/ansilove.rb (hunk 4 above)
Comment (usbsnowcrash) [reply Formula/ansilove.rb]: Done

File: Formula/ansilove.rb (hunk 5 above)
Comment (usbsnowcrash) [reply Formula/ansilove.rb]: Done

File: Formula/ansilove.rb (hunk 3 above)
Comment (usbsnowcrash) [reply Formula/ansilove.rb]: Changed

**Comment by usbsnowcrash** [comment]
@fxcoudert I think I addressed all your feedback, let me know if it looks ok 💛

File: Formula/ansilove.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  url "https://github.com/ansilove/ansilove/archive/3.0.8.tar.gz"
+  sha256 "655472d31a9ec5a5ba29f5c15485d8647526ff8b0c9968d71f4267a9aa9bb544"
+
+  depends_on "cmake" => :build
+  depends_on "gd"
+
+  def install
+    system "cmake", "."
+    system "make"
+    bin.install "ansilove"
```
Comment (fxcoudert) [inline Formula/ansilove.rb]: Is there a `make install` target that could do that for us?

File: Formula/ansilove.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  test do
+    system "#{bin}/ansilove", "-v"
```
Comment (fxcoudert) [inline Formula/ansilove.rb]: We need a functional test, that does something and check the output.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/ansilove.rb (hunk 7 above)
Comment (usbsnowcrash) [reply Formula/ansilove.rb]: What is a functional test for a utility like this?  Should I wget an ansi file from somewhere and try to convert it to png (that's what this utility does).

File: Formula/ansilove.rb (hunk 6 above)
Comment (usbsnowcrash) [reply Formula/ansilove.rb]: Sorry I'm not much of a C developer.  There is a make install but I always get file permission errors when I do that locally?  I'm guessing I need to set the correct install destination but reading the Makefile I don't really understand how to do that :(

output when I use make install
```
CMake Error at cmake_install.cmake:36 (file):
  file INSTALL cannot copy file
  "/tmp/ansilove-20180409-93980-1hm6m3i/ansilove-3.0.8/ansilove" to
  "/usr/local/bin/ansilove".
```

File: Formula/ansilove.rb (hunk 7 above)
Comment (fxcoudert) [reply Formula/ansilove.rb]: Can you install one of the example ASCII files, convert it to PNG and check that the PNG file has indeed been created?

**Pull request closed without merging**